src/lib/arch/aarch64/InstructionMetadata.hh

```diff
@@ -84,6 +84,12 @@ struct InstructionMetadata {
   /** The number of explicit operands. */
   uint8_t operandCount;
 
+  /** The current exception state of this instruction. */
+  InstructionException metadataException_ = InstructionException::None;
+
+  /** Whether an exception has been encountered. */
+  bool metadataExceptionEncountered_ = false;
```

Comment:
These should be private with appropriate getters given they should only be modified by `InstructionMetadata.cc`